source/grid/grid_tools.cc

```diff
@@ -5962,6 +5962,39 @@ namespace GridTools
 
   namespace internal
   {
+    std::tuple<std::vector<unsigned int>,
+               std::vector<unsigned int>,
+               std::vector<unsigned int>>
+    convert_guessed_ranks_and_indices_to_crs(
+      std::vector<std::pair<unsigned int, unsigned int>> &ranks_and_indices)
+    {
+      std::sort(ranks_and_indices.begin(), ranks_and_indices.end());
+
+      std::vector<unsigned int> ranks;
+      std::vector<unsigned int> ptr;
+      std::vector<unsigned int> indices;
+
+      unsigned int dummy_rank = numbers::invalid_unsigned_int;
+
+      for (const auto &i : ranks_and_indices)
```

Comment:
I don't like the `auto` keyword when it makes it difficult to see the underlying type. The function here is short so this is borderline, but I still prefer to spell it out for clarity reasons.
```suggestion
      for (const std::pair<unsigned int, unsigned int> &i : ranks_and_indices)
```